src/modules/cpu/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,127 @@
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include <omp.h>
+
+Rpp32f getSquare(Rpp32f &value)
+{
+    return (value * value);
+}
+
+RppStatus non_silent_region_detection_host_tensor(Rpp32f *srcPtr,
+                                                  RpptDescPtr srcDescPtr,
+                                                  Rpp32s *srcLengthTensor,
+                                                  Rpp32f *detectedIndexTensor,
+                                                  Rpp32f *detectionLengthTensor,
+                                                  Rpp32f cutOffDB,
+                                                  Rpp32s windowLength,
+                                                  Rpp32f referencePower,
+                                                  Rpp32s resetInterval,
+                                                  rpp::Handle& handle)
+{
+    Rpp32u numThreads = handle.GetNumThreads();
+    const Rpp32f cutOff = std::pow(10.0f, cutOffDB * 0.1f);
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        Rpp32s srcLength = srcLengthTensor[batchCount];
+
+        // mmsBuffer length is equal to input audio length and can vary dynamically for each input in a batch
+        // preallocating a static buffer for entire batchsize will be too big, so allocate mmsBuffer for each sample dynamically
+        Rpp32f *mmsBuffer = static_cast<Rpp32f *>(calloc(srcLength, sizeof(Rpp32f)));
+        bool referenceMax = (referencePower == 0.0f);
+
+        // set reset interval based on the user input
+        Rpp32s resetLength = (resetInterval == -1) ? srcLength : resetInterval;
+
+        // calculate moving mean square of input
+        Rpp32f meanFactor = 1.0f / windowLength;
+        Rpp32s windowBegin = -windowLength + 1;
+        for (Rpp32s outPos = 0; outPos < srcLength;)
+        {
+            Rpp32f sumOfSquares = 0.0f;
+            for (Rpp32s i = std::max<Rpp32s>(windowBegin, 0); i < outPos; i++)
```

Comment:
not sure how this works with the loop from line #63. I don't understand what you are trying to do here

Reply:
Hi Rajy,
I have given an explanation for this code in below document with an example
https://docs.google.com/document/d/1OuzIl_DN_U_GAbhr4seVWUQETs2rEVGTWkC5WyiLHCo/edit?usp=sharing

Please check this document and we can have a call if needed